### PR TITLE
chore: refactor code to remove config.EnvNameNone

### PR DESCRIPTION
**Pull request by Louise15926**
This PR refactors codes to remove config.EnvNameNone. Packages affected include: term/selector, cli, config, deploy/cloudformation/stack. 

Related [this comment in #1101 ](https://github.com/aws/copilot-cli/pull/1101#discussion_r452521317).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/term/selector/selector.go [hunk 1]
```diff
@@ -265,21 +265,18 @@ func (s *Select) Environment(prompt, help, app string) (string, error) {
 	return selectedEnvName, nil
 }
 
-// EnvironmentWithNone fetches all the environments in an app and prompts the user to select one of the environments or None.
-func (s *Select) EnvironmentWithNone(prompt, help, app string) (string, error) {
+// EnvironmentWithAdditionalOptions fetches all the environments in an app and prompts the user to select one of the environments or additional options.
+// NOTE: `additionalOpt` shouldn't be an empty string.
+func (s *Select) EnvironmentWithAdditionalOptions(prompt, help, app string, additionalOpt string, moreAdditionalOpts ...string) (string, error) {
```
Comment (iamhopaul123) [inline internal/pkg/term/selector/selector.go:270]: Got a bit confused at this one. Why not just
```suggestion
func (s *Select) EnvironmentWithAdditionalOptions(prompt, help, app string, additionalOpt ...string) (string, error) {
```

File: internal/pkg/term/selector/selector.go [hunk 2]
```diff
@@ -265,21 +265,18 @@ func (s *Select) Environment(prompt, help, app string) (string, error) {
...
 	if err != nil {
 		return "", err
 	}
 
-	if len(envs) == 0 {
-		log.Infof("No environment found associated with app %s, defaulting to %s (task will run in your default VPC)\n",
-			color.HighlightUserInput(app), color.Emphasize(config.EnvNameNone))
-		return config.EnvNameNone, nil
+	for _, opt := range append(moreAdditionalOpts, additionalOpt) {
+		envs = append(envs, opt)
```
Comment (iamhopaul123) [inline internal/pkg/term/selector/selector.go:277]: And here
```suggestion
	for _, opt := range additionalOpt {
		envs = append(envs, opt)
```

File: internal/pkg/term/selector/selector.go (hunk 1 above)
Comment (Louise15926) [reply internal/pkg/term/selector/selector.go]: Indeed they can be combined. I wanted to have custom log information for "environmentwithadditionaloptions" but that no longer seems necessary now.

File: go.sum [hunk 3]
```diff
@@ -142,6 +142,7 @@ github.com/gobuffalo/logger v1.0.3 h1:YaXOTHNPCvkqqA7w05A4v0k2tCdpr+sgFlgINbQ6gq
 github.com/gobuffalo/logger v1.0.3/go.mod h1:SoeejUwldiS7ZsyCBphOGURmWdwUFXs0J7TCjEhjKxM=
 github.com/gobuffalo/packd v1.0.0 h1:6ERZvJHfe24rfFmA9OaoKBdC7+c9sydrytMg8SdFGBM=
 github.com/gobuffalo/packd v1.0.0/go.mod h1:6VTc4htmJRFB7u1m/4LeMTWjFoYrUiBkU9Fdec9hrhI=
+github.com/gobuffalo/packr v1.30.1 h1:hu1fuVR3fXEZR7rXNW3h8rqSML8EVAf6KNm0NKO/wKg=
```
Comment (efekarakus) [inline go.sum:145]: intersting, why did this come? since we have packr/v2 already.

Can you run "go mod tidy" to be sure

Comment (Louise15926) [reply go.sum]: it goes away after `go mod tidy`!

**Pull request merged**